### ISSUE TITLE
Home Assistant OS 2022.7.6 - No data from the EMU-2

### ISSUE THREAD

**Issue by acato** [ryanwinter/hass-rainforest-emu-2#22]
The integration took a few tries to install (it was hanging and crashing Home Assistant). Eventually it worked, and I can see the total consumption but all other sensors are "unknown" and don't update.

**Comment by KTibow** [comment]
Same. This is what I see:
![image](https://user-images.githubusercontent.com/10727862/180813593-2fcd619e-3d58-46f7-9bc5-0f02c42371e6.png)

**Comment by KTibow** [comment]
Here's my (filtered) [home-assistant.log](https://github.com/ryanwinter/hass-rainforest-emu-2/files/9182547/home-assistant.log), @ryanwinter would you mind to take a look at it?

**Comment by HyperActiveJ** [comment]
Did this happen start after an update or is 2022.7.6 the first version of HA you have tried this integration on?

**Comment by HyperActiveJ** [comment]
> Here's my (filtered) [home-assistant.log](https://github.com/ryanwinter/hass-rainforest-emu-2/files/9182547/home-assistant.log), @ryanwinter would you mind to take a look at it?

Your log seems to indicate that
1. the EMU2 is trying to connect to the meter:
`<Status>Rejoining</Status>`
2. Signal strength on all channels appears to be 0
`<LinkStrength>0x00</LinkStrength>`
3. The only usage data it has reported is basically 0 given the devisor, which also seems invalid:
```
<CurrentUsage>0x0000000000000002</CurrentUsage>
<Divisor>0x000003e8</Divisor>
```

I would check on the device to confirm its really reporting something from the meter. If you see real(ish) time data on the display then report back and I'll scratch my head somemroe.

**Comment by KTibow** [comment]
Yeah it ended up being too far from the meter, so now I'm trying the TCP thing. Would you mind taking a look at that?

**Comment by HyperActiveJ** [comment]
> Yeah it ended up being too far from the meter, so now I'm trying the TCP thing. Would you mind taking a look at that?

Open an issue for the tcp connection causing HA to crash when the remote end kills the connection prior to sending data. We can take both the conversion over to that issue. I'll try to commit a fix to my git hub tonight. You'll need to install the component manually, not via hacs, to test it. Can you do that?
